packages/accounts-controller/src/AccountsController.ts

```diff
@@ -0,0 +1,607 @@
+    });
+  }
+
+  /**
+   * Updates the internal accounts list by retrieving legacy and snap accounts,
+   * removing duplicates, and updating the metadata of each account.
+   *
+   * @returns A Promise that resolves when the accounts have been updated.
+   */
+  async updateAccounts(): Promise<void> {
+    let legacyAccounts = await this.#listLegacyAccounts();
+    let snapAccounts: InternalAccount[] = [];
+    if (this.keyringApiEnabled) {
+      snapAccounts = await this.#listSnapAccounts();
+      // remove duplicate accounts that are retrieved from the snap keyring.
+      legacyAccounts = legacyAccounts.filter(
+        (account) =>
+          !snapAccounts.find(
+            (snapAccount) => snapAccount.address === account.address,
+          ),
+      );
+    }
+
+    // keyring type map.
+    const keyringTypes = new Map<string, number>();
+    const previousAccounts = this.state.internalAccounts.accounts;
+
+    const accounts: Record<string, InternalAccount> = [
+      ...legacyAccounts,
+      ...snapAccounts,
+    ].reduce((internalAccountMap, internalAccount) => {
+      const keyringTypeName = keyringTypeToName(
+        internalAccount.metadata.keyring.type,
+      );
+      const keyringAccountIndex = keyringTypes.get(keyringTypeName) ?? 0;
+      if (keyringAccountIndex) {
+        keyringTypes.set(keyringTypeName, keyringAccountIndex + 1);
+      } else {
+        keyringTypes.set(keyringTypeName, 1);
+      }
+
+      const existingAccount = previousAccounts[internalAccount.id];
+
+      internalAccountMap[internalAccount.id] = {
+        ...internalAccount,
+
+        metadata: {
+          ...internalAccount.metadata,
+          name:
+            existingAccount && existingAccount.metadata.name !== ''
+              ? existingAccount.metadata.name
+              : `${keyringTypeName} ${keyringAccountIndex + 1}`,
+          lastSelected: existingAccount?.metadata?.lastSelected,
+        },
+      };
+
+      return internalAccountMap;
+    }, {} as Record<string, InternalAccount>);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore Type instantiation is excessively deep and possibly infinite.
+    this.update((currentState: AccountsControllerState) => {
+      currentState.internalAccounts.accounts = accounts;
+    });
+  }
+
+  /**
+   * Loads the backup state of the accounts controller.
+   *
+   * @param backup - The backup state to load.
+   */
+  loadBackup(backup: AccountsControllerState): void {
+    if (backup.internalAccounts) {
+      this.update((currentState: AccountsControllerState) => {
+        currentState.internalAccounts = backup.internalAccounts;
+      });
+    }
+  }
+
+  /**
+   * Returns a list of internal accounts created using the SnapKeyring.
+   *
+   * @returns A promise that resolves to an array of InternalAccount objects.
+   */
+  async #listSnapAccounts(): Promise<InternalAccount[]> {
+    // const [snapKeyring] = this.getKeyringByType(SnapKeyring.type);
```

Comment:
Dead code

Reply:
removed in c237821dd9f4931117742bbb830347102c89402b